bin/check-package-metadata.js

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env node
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: loopback-next
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+/**
+ * This is an internal script that will verify that all packages in our monorepo
+ * are registered in all required places and have the expected metadata in their
+ * package.json.
+ */
+'use strict';
+
+const path = require('path');
+const fs = require('fs-extra');
+
+const Project = require('@lerna/project');
+
+/**
+ * Check existence of LICENSE file in the monorepo packages
+ * @param {Package[]} packages A list of @lerna/project packages
+ */
+async function checkLicenseFiles(packages) {
+  return new Promise(resolve => {
```

Comment:
`new Promise` is a kind of an anti-pattern, the only place where it's justified is when you need to wrap callback-based APIs.

```ts
async function checkLicenseFiles(packages) {
  const errors = [];
  for (const p of packages) {
    if (!fs.existsSync(path.join(p.location, 'LICENSE'))) {
      errors.push(`${p.name} directory doesn't contain LICENSE file`);
    }
   }
  return errors;
}
```

Your current implementation is not async, so you can simply change `async function` to a regular `function`. 

However, I think it would be better to check the files in parallel to get better performance.

```ts
function checkLicenseFiles(packages) {
  return Promise.all(packages.map(async (p) => {
    const exists = await fs.pathExists(path.join(p.location, 'LICENSE'));
    return exists ? undefined : `${p.name} directory doesn't contain LICENSE file`
  });
}
```

Reply:
Agree

Reply:
@raymondfeng @bajtos Let's finalize the decision, please)

Reply:
I'm fine either way.